### ISSUE TITLE
__slots__ declaration ignored in a subclass of generic collections.abc.AsyncIterator

### ISSUE THREAD

**Issue by dmoklaf** [PyCQA/pylint#5099]
### Bug description

In Python 3.9, as part of the annotation improvements, collections.abc.AsyncIterator has replaced typing.AsyncIterator for annotations.

However, this replacement has broken down pylint for this atiterator.py file (which had no pylint issues before and has no runtime issues):

--------------------------------------
```python
"Bla bla bla"

import asyncio
import collections
import typing

Type = typing.TypeVar("Type")

# pylint: disable=R0903
class AsyncTimedIterator(collections.abc.AsyncIterator[Type]):
    "Bla bla bla"

    __slots__ = ("_iterator", "_timeout")

    def __init__(
        self, iterator: collections.abc.AsyncIterator[Type], timeout: float
    ) -> None:
        self._iterator = iterator
        self._timeout = timeout

    async def __anext__(self) -> Type:
        return await asyncio.wait_for(self._iterator.__anext__(), self._timeout)
```

### Command used

```shell
pylint atiterator.py
```

### Pylint output

```shell
************* Module extract
aiterator.py:18:8: E0237: Assigning to attribute '_iterator' not defined in class slots (assigning-non-slot)
aiterator.py:19:8: E0237: Assigning to attribute '_timeout' not defined in class slots (assigning-non-slot)

-------------------------------------------------------------------
Your code has been rated at 0.91/10 (previous run: 10.00/10, -9.09)
```

### Expected behavior

No E0237 warning

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.0
Python 3.9.5 (default, May 18 2021, 12:31:01) 
[Clang 10.0.0 ]
MacOS 11.6
```

### OS / Environment

MacOS 11.6
conda 4.10.3
zsh 5.8 (x86_64-apple-darwin20.0)

### Additional dependencies

(Available by checking the edition history)

**Comment by Pierre-Sassoulas** [comment]
This might be fixed if you upgrade astroid or on a new install, because we released astroid 2.12.13 with https://github.com/PyCQA/astroid/pull/1861 and pylint can use the latest astroid. Keeping in 2.15.7, do you think we need a regression tests @mbyrnepr2 ?

**Comment by mbyrnepr2** [comment]
I'm thinking no @Pierre-Sassoulas because of the test in astroid & based on the conversation around duplicating tests in Pylint which are already in astroid.
But let me know & I'm happy to add a test here if there are some doubts with the testing 🧪
